js/plugin/RouteLoaderConverter.js

```diff
@@ -0,0 +1,417 @@
+			}); 
+			$('#loadedittrackdlg').on('show.bs.modal', (e) => { 
+				$('#manual_collapse').collapse('hide');
+				this._closeCanceled=true;
+			});
+			
+			L.DomUtil.get('submitLoadEditTrack').onclick = L.bind(function(){
+				this._closeCanceled=false;
+				this.onBusyChanged(true);
+				if(this._testLayer.getLayers().length > 0)
+				{
+					this._testLayer.clearLayers();
+					setTimeout(() => {
+						this.addRoutingPoints();
+						this.onBusyChanged(false);
+						$('#loadedittrackdlg').modal('hide');
+					},0);
+				}
+				else
+					setTimeout(() => {
+						this.convertTrackLocal();
+						$('#loadedittrackdlg').modal('hide');
+					},0);
+
+				
+			}, this);
+			
+			L.DomUtil.get('simplify_tolerance').onchange = L.bind(this.onToleranceSlider, this);
+			
+			L.DomUtil.get('loadedittrackFile').onchange = L.bind(this.onFileChanged, this);
+			this.onFileChanged({target: L.DomUtil.get('loadedittrackFile') });
+			
+			this.setDialogDraggable($("#loadedittrackdlg .modal-header"));
+			
+			$('#manual_collapse').collapse('hide');
+			$('#manual_collapse').on('hidden.bs.collapse shown.bs.collapse',  (e) => {this.onManualCollapse(e);});
+			
+			// dummy, no own representation, delegating to EasyButton
+			let dummy= L.DomUtil.create('div');
+			dummy.hidden = true;
+            return dummy;
+		},
+
+		onRemove: function(map) {
+			// Nothing to do here
+		},
+		
+		onFileChanged: function(e) {
+			if(!e.target.files[0]) return;
+			$(e.target).next("label").text(e.target.files[0].name);
+			let testmode=this._options.isTestMode;
+			this.cleanup();
+			this._options.isTestMode=testmode;
+			if(this._options.isTestMode)
+			{
+				this.once("file:loaded",this.setSliderRange,this)
+				this.convertTrackLocal();
+			}
+		},
+		
+		setLayerNameFromGeojson: function(geoJSON) {
+			if(geoJSON.type == "Feature" && geoJSON.properties && geoJSON.properties.name)
+			{
+				this._layerName=geoJSON.properties.name;
+				return;
+			}
+			
+			if(geoJSON.type == "FeatureCollection")
+			{
+				for(let i=0; i < geoJSON.features.length; i++)
+				{
+					if(geoJSON.features[i].properties && geoJSON.features[i].properties.name)
+					{
+						this._layerName=geoJSON.features[i].properties.name;
+						return;
+					}
+				}
+				
+			}
+		},
+		
+		getOptions: function() {
+			this._options.showTrackLayer = $("#cb_showtracklayer")[0].checked;
+			this._options.showPointAsPoi = $("#cb_showpois")[0].checked;
+
+			this._options.simplifyTolerance = -1;
+			this._bounds=undefined;
+
+			let f=$("#file_format").val();
+			this._options.format=(f == "extension")?undefined:f;
+
+		},
+		
+		convertTrackLocal: function(){
+		   this.onBusyChanged(true);
+			
+			this.getOptions();
+			
+            if (toGeoJSON === undefined) {
+                if (typeof window !== 'undefined') {
+                    toGeoJSON = require('togeojson');
+                } else {
+                    toGeoJSON = require('togeojson')(root);
+                }
+            }

```

Comment:
We don't need this? Could you move this to external code when using in other contexts?